protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java

```diff
@@ -125,46 +125,38 @@ private static void exportResource(String resourceName, ClassLoader classLoader)
     }
     
     @Override
-    public void execute() {
-        //Help flag check
-        if (helpFlag) {
-            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(getName());
-            outStream.println(commandUsageInfo);
-            return;
-        }
+ public void execute() {
+    // Help flag check
+    if (helpFlag) {
+        String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(getName());
+        outStream.println(commandUsageInfo);
+        return;
+    }
 
+    try {
         if (protoPath.endsWith("**.proto") || new File(protoPath).isDirectory()) {
             // Multiple proto files
-            List<String> protoFiles;
-            try {
-                protoFiles = getProtoFiles(protoPath);
-            } catch (IOException e) {
-                String errorMessage = "Failed to find proto files in the directory. " +
-                        "Please input a valid proto files directory.";
-                outStream.println(errorMessage);
-                return;
+            List<String> protoFiles = getProtoFiles(protoPath);
+            if (protoFiles.isEmpty()) {
+                throw new RuntimeException("Input directory does not contain any proto files. " +
+            "Please input a valid proto files directory."); 
             }
-            if (protoFiles.size() == 0) {
-                String errorMessage = "Input directory does not contain any proto files. " +
-                        "Please input a valid proto files directory.";
-                outStream.println(errorMessage);
-            } else {
-                for (String protoFile : protoFiles) {
-                    generateBalFile(protoFile);
-                }
+            for (String protoFile : protoFiles) {
+                generateBalFile(protoFile);
             }
         } else {
             // Single proto file
-            // check input protobuf file path
             Optional<String> pathExtension = getFileExtension(protoPath);
             if (pathExtension.isEmpty() || !PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
-                String errorMessage = "Invalid proto file path. Please input valid proto file location.";
-                outStream.println(errorMessage);
-                return;
+                throw new RuntimeException("Input directory does not contain any proto files. " +
+            "Please input a valid proto files directory.");
             }
             generateBalFile(protoPath);
         }
+    } catch (Exception e) {
```

Comment:
Even if we do a catch here, the error is logged elsewhere and also printed to the stdout. The exception is not thrown to here. Hence, this catch will not be able to catch it. We need to remove the log with the stacktrace and display a reason for the failure if possible.

Reply:
@dilanSachi

Reply:
@Abhi132004. It seems you have not understood what I meant in my comment. Let me try and explaing a bit more. The issue is related to the `generateBalFile` method. In this method, when there are errors in the descriptor generator phase, they are caught as a `CodeGeneratorException` in here [1]. However, we have only printed a generic error and the error message. But this prints stack traces as well. What we need to do is to check if we can add any error information to this printed error and try and stop the stack trace printing in the cmd.

[1] https://github.com/ballerina-platform/protoc-tools/blob/6db60af220d182506a59f49d8d527ccaa4953b4d/protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java#L265